### ISSUE TITLE
ArmingState 10214

### ISSUE THREAD

**Issue by austinmroczek** [craigjmidwinter/total-connect-client#115]
While investigating #114 the user's new ProA7 panel returned state 10214 which we don't know.

Need help to understand this state, and any other new state codes related to this panel.

**Comment by austinmroczek** [comment]
@rootdet I'd ask for your help here too please.  I just noticed your panel said it's ArmingState (armed-away, disarmed, etc) is code 10214.  Our code doesn't know what that means.  You can see the other codes in the ALARM_STATUS.md file

**Comment by rootdet** [comment]
If you can proceed me a simple way to extract the code, i can go through the various arming, disarms, and troubles.

**Comment by austinmroczek** [comment]
This will be the most annoying of tests I ask for because it involves arming/disarming/alarming the system, waiting long enough to get the return from the info.py script, and then disarming.  

We might want to wait on this one until we get a bit further along with the partition support, because there's a chance that could impact the responses we get.

**Comment by austinmroczek** [comment]
That said, if you could confirm that your system was indeed Disarmed when you sent the original log file, that's a start.

**Comment by dulitz** [comment]
Unfortunately I lost the logs when I accidentally bounced my container, but I'm pretty sure I have seen 10214 out of my (older) panel as well. It was somewhere during a sequence of "heat detector zone trouble," "heat detector zone alarm," "alarm silenced (but not cleared)," and "alarm cleared."

All security zones were disarmed during that time, though the heat detector zone was obviously not disarmed. And of course I could be mistaken about 10214 -- it was an alarm code unknown to your codebase though.

**Comment by rootdet** [comment]
When the logs were sent, the panel was disarmed but zones would have been in trouble. I would have had at least a window open and possibly a door open. In the P2 Area, you would have soon potentially both sensors tripped Motion+door

**Comment by austinmroczek** [comment]
At least for all panels I've seen to date, individual zones being "faulted" (like an open door, motion detected) doesn't change the ArmingState of the panel.  Trouble is a different state for a zone, usually low battery, but it also doesn't affect ArmingState that I've seen to date.  

To help enumerate the codes, just use `info.py` to get the state of your system.  Under the Location you'll see Arming State listed.  

Start with the panel disarmed with all zones in good condition.  Run info.py, record state.

Do the same for armed stay, armed away, etc.  The most annoying to test are those states that trigger the siren - pushing the test button on a smoke detector, triggering the alarm, etc.  It's also annoying if your development computer is far from the panel...I had to run up/down stairs with the alarm blaring.

**Comment by dulitz** [comment]
So I'm not running this in HA, I'm just dumping it into a Prometheus database. I've just raised my retention from 15 days to 2 years, so next time I get an unusual state I'll have everything...

Because we have been having some alarm malfunctions, our test state is indeterminate at the moment, so I won't be doing tests this weekend. Hopefully they will fix it next week.

**Comment by austinmroczek** [comment]
I just realized there is a difference between what the Location reports as ArmingState and what the Partition reports.  Mine is only a single partition, but here's what I see from my Lynx Touch 7000:

Location ArmingState | Partition 1 ArmingState | Note
------------ | - | - 
10226 | 10203 | Armed Stay
10226 | 10307 | Arming
10214 | 10200 | Disarmed
10214 | 10200 | Disarmed with zone faulted

I think when I opened this issue I was looking at the Location ArmingState, which at the moment does not matter for our code.  Assuming the ProA7 returns the same type of Partition ArmingState codes we'll be o.k.  If not we'll need to re-open this.
